pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -332,34 +322,72 @@ func (ia *IngressAnalyzer) getIngressObjectTargetedPeersAndPorts(ns string,
+			if err != nil {
+				return nil, err
 			}
-			permittedPeerConn := common.MakeConnectionSet(false)
-			// check if its TCP conns contains the required port
-			peerTCPConn := eval.GetPeerExposedTCPConnections(peer)
-			if peerTCPConn.Contains(strconv.Itoa(portNum), string(corev1.ProtocolTCP)) {
-				permittedPort := common.PortSet{}
-				permittedPort.AddPort(intstr.FromInt(portNum))
-				permittedPeerConn.AddConnection(corev1.ProtocolTCP, permittedPort)
-				if _, ok := res[peer]; !ok {
-					res[peer] = permittedPeerConn
-				} else {
-					res[peer].Union(permittedPeerConn)
-				}
+			if _, ok := res[peer]; !ok {
+				res[peer] = currIngressPeerConn
+			} else {
+				res[peer].Union(currIngressPeerConn)
 			}
 		}
 	}
 	return res, nil
 }
+
+// getIngressPeerConnection returns the ingress connection to a peer based on the required port specified in the ingress objects
+func (ia *IngressAnalyzer) getIngressPeerConnection(peer eval.Peer, actualServicePorts []corev1.ServicePort,
+	requiredPort intstr.IntOrString) (*common.ConnectionSet, error) {
+	peerTCPConn := eval.GetPeerExposedTCPConnections(peer)
+	// if the required port is not specified - all peer's TCP connections are allowed for ingress
+	if requiredPort.IntVal == 0 && requiredPort.StrVal == "" {
+		return peerTCPConn, nil
+	}
+	// get the peer port which may be accessed by the service required port
+	peerPortToFind := getPeerAccssesPort(actualServicePorts, requiredPort)
+	// compute the connection to the peer with the required port
+	res := common.MakeConnectionSet(false)
+	portNum := peerPortToFind.IntValue()
+	if peerPortToFind.StrVal != "" { // if the port we are searching for is namedPort
+		portInt, err := ia.pe.ConvertPeerNamedPort(peerPortToFind.StrVal, peer)
+		if err != nil {
+			return nil, err
+		}
+		portNum = int(portInt)
+	}
+
+	if peerTCPConn.Contains(strconv.Itoa(portNum), string(corev1.ProtocolTCP)) {
+		permittedPort := common.PortSet{}
+		permittedPort.AddPort(intstr.FromInt(portNum))
+		res.AddConnection(corev1.ProtocolTCP, permittedPort)
+	}
+	return res, nil
+}
+
+// getPeerAccssesPort returns the peer's port to be exposed based on the service's port.targetPort value
+func getPeerAccssesPort(actualServicePorts []corev1.ServicePort, requiredPort intstr.IntOrString) intstr.IntOrString {
+	var svcPodAccessPort intstr.IntOrString
+	// get the peer port to find from the required port
+	for _, svcPort := range actualServicePorts {
+		// extracting the pod access port from the service port
+		if !(svcPort.TargetPort.IntVal == 0 && svcPort.TargetPort.StrVal == "") {
+			svcPodAccessPort = svcPort.TargetPort
+		} else {
+			svcPodAccessPort.IntVal = svcPort.Port
+		}
+		// checking if the service port matches the required port, if yes returning its pod access port
+		switch {
+		case svcPort.Name != "" && svcPort.Name == requiredPort.StrVal, // or
+			svcPort.Port == requiredPort.IntVal, // or
+			svcPort.TargetPort == requiredPort:
+			return svcPodAccessPort
+		}
+	}
```

Comment:
the service target port should match the pod's port

Reply:
added explaination - copied from [here](https://pkg.go.dev/k8s.io/api@v0.26.1/core/v1#ServicePort.TargetPort)